methods/inputs/locate_gedi_data.py

```diff
@@ -103,48 +124,106 @@ def gedi_fetch(boundary_file: str, gedi_data_dir: str) -> None:
             working_layer.CreateFeature(new_feature)
-            del destination_data_source # aka destination_data_source.Close()
+            del destination_data_source  # close the GeoJSON file
 
+            # read the GeoJSON as a GeoPandas dataframe for further processing
             shape = gpd.read_file(chunk_path)
 
-            # We set max_coords rather low (the upper bound is less than 5000) as some of
-            # our geometries are causing issues with being too precise and the coordinates
-            # not all being > 1m apart. Checking this for every geometry is a bit hard, so
-            # instead we approximate it by setting the bar for simplification much lower.
+            # simplify the shape to avoid excessive precision issues in coordinates
             shape = check_and_format_shape(shape, simplify=True, max_coords=200)
+
+            # query the GEDI data for this shape and specified time range
             result = query(
                 product=GediProduct.L4A,
-                date_range=(dt.datetime(2020, 1, 1, 0, 0), dt.datetime(2021, 1, 1, 0, 0)),
+                date_range=(dt.datetime(start_year, 1, 1), dt.datetime(end_year, 1, 1)),
                 spatial=shape
             )
             granule_metadatas.append(result)
 
+            # move to the next chunk
             chunk = chunked_layer.GetNextFeature()
 
+    # concatenate all metadata into a single dataframe and remove duplicates
     granule_metadata = pd.concat(granule_metadatas).drop_duplicates(subset="granule_name")
 
+    # write the granule names to a text file in the specified directory
+    granule_names = granule_metadata["granule_name"].tolist()
+    output_txt_path = os.path.join(output_txt_dir, "gedi_names.txt")
+    with open(output_txt_path, "w") as txt_file:
+        for name in granule_names:
+            txt_file.write(f"{name}\n")
+
+    # save each granule's metadata as a JSON file in the specified GEDI data directory
     for _, row in granule_metadata.iterrows():
         basename, _ = os.path.splitext(row.granule_name)
         with open(os.path.join(gedi_data_dir, f"{basename}.json"), "w", encoding='utf-8') as f:
-            f.write(json.dumps(
+            json.dump(
                 {
                     "name": row.granule_name,
                     "url": row.granule_url
-                }
-            ))
+                },
+                f
+            )
 
+
+# main function to handle input arguments and call the data-fetching function
 def main() -> None:
-    try:
-        boundary_file = sys.argv[1]
-        gedi_data_dir = sys.argv[2]
-    except IndexError:
-        print(f"Usage: {sys.argv[0]} BUFFER_BOUNDRY_FILE GEDI_DATA_DIRECTORY")
-        sys.exit(1)
-    except DownloadError as exc:
-        print(f"Failed to download: {exc.msg}")
-        sys.exit(1)
-
-    gedi_fetch(boundary_file, gedi_data_dir)
+    # set up argument parsing
+    parser = argparse.ArgumentParser(
```

Comment:
Thanks for bringing this script inline with the others!

I suspect this does mean again scripts/tmfpython.sh will need updating

Reply:
Yes, thats right, I'll make the edits to the calls for both `locate_gedi_data.py `and `filter_gedi_data.py` in the shellscript